pyigm/fN/fnmodel.py

```diff
@@ -0,0 +1,456 @@
+      'Gamma' -- Gamma function [Following Inoue+14]
+    zmnx : tuple
+      Redshift range where this model applies (zmin,zmax)
+    pivots : array
+          log NHI values for the pivots
+    param : ndarray
+      Parameter array
+    zpivot : float, optional
+          Pivot for redshift evolution (2.4)
+    gamma : float, optional
+          Power law for dN/dX, not dN/dz (1.5)
+    """
+    @classmethod
+    def default_model(cls, use_mcmc=False):
+        """ Pass back a default fN_model from Prochaska+14
+
+        Tested against XIDL code by JXP on 09 Nov 2014
+
+        Parameters
+        ----------
+        recalc : boolean, optional (False)
+          Recalculate the default model
+        use_mcmc : boolean, optional (False)
+          Use the MCMC chain to generate the model
+        write : boolean, optional (False)
+          Write out the model
+        """
+        if use_mcmc:
+            # MCMC Analysis (might put these on a website)
+            chain_file = (os.environ.get('DROPBOX_DIR')+
+                        'IGM/fN/MCMC/mcmc_spline_k13r13o13n12_8.fits.gz')
+            outp = mcmc.chain_stats(chain_file)
+
+            # Build a model
+            NHI_pivots = [12., 15., 17.0, 18.0, 20.0, 21., 21.5, 22.]
+            fN_model = cls('Hspline', zmnx=(0.5,3.0),
+                            pivots=NHI_pivots, param=outp['best_p'])
+        else:
+            # Input the f(N) at z=2.4 from Prochaska+13
+            fN_file = (pyigm_path+'/data/fN/fN_spline_z24.fits.gz')
+            print('Using P14 spline values to generate a default model')
+            print('Loading: {:s}'.format(fN_file))
+            hdu = fits.open(fN_file)
+            fN_data = hdu[1].data
+            # Instantiate
+            fN_model = cls('Hspline', zmnx=(0.5, 3.0),
+                            pivots=np.array(fN_data['LGN']).flatten(),
+                            param=np.array(fN_data['FN']).flatten())
+        # Return
+        return fN_model
+
+    def __init__(self, mtype, zmnx=(0., 0.), pivots=None, param=None,
+                 zpivot=2.4, gamma=1.5):
+        self.mtype = mtype  # Should probably check the choice
+
+        # Inoue+14
+        if mtype == 'Gamma':
+            zmnx = (0., 10.)
+            param = [ [12., 23., 21., 28.], # Common
+                      [1.75828e8, 9.62288e-4],             # Bi values
+                      [500, 1.7, 1.2, 4.7, 0.2, 2.7, 4.5], # LAF
+                      [1.1, 0.9, 2.0, 1.0, 2.0] ] # DLA
+        self.zmnx = zmnx  
+
+        # Pivots
+        if pivots is None: 
+            self.pivots = np.zeros(2)
+        else: 
+            self.pivots = pivots
+        self.npivot = len(self.pivots)
+
+        # Param
+        if param is None:
+            self.param = np.zeros(self.npivot)
+        else:
+            self.param = param
+            # Init
+            if mtype == 'Hspline':
+                self.model = scii.PchipInterpolator(self.pivots, self.param,
```

Comment:
I guess this works, but for code readability  and future expansions, I would prefer to see the "if mtype =="  all at the same  level of indentation (and perhaps one after the other). I would therefore suggest a structure like this:

if pivots:
if param:
if mtype:

and then handle the special cases.